include/flox/resolver/descriptor.hh

```diff
@@ -0,0 +1,182 @@
+ */
+struct ManifestDescriptorRaw {
+
+  public:
+
+    /** Match `name`, `pname`, or `pkgAttrName` */
+    std::optional<std::string> name;
+
+    /** Match `version` or `semver` if a modifier is present. */
+    std::optional<std::string> version;
+
+    /** Match a catalog stability. */
+    std::optional<std::string> stability;
+
+    /** @brief A dot separated attribut path, or list representation. */
+    using Path = std::variant<std::string, flox::AttrPath>;
+    /** Match a relative path. */
+    std::optional<Path> path;
+
+    /**
+     * @brief A dot separated attribut path, or list representation.
+     *        May contain `null` members to represent _globs_.
+     */
+    using AbsPath = std::variant<std::string, AttrPathGlob>;
+    /** Match an absolute path, allowing globs for `system`. */
+    std::optional<AbsPath> absPath;
+
+    /** Only resolve for a given set of systems. */
+    std::optional<std::vector<std::string>> systems;
+
+    /** Whether resoution is allowed to fail without producing errors. */
+    std::optional<bool> optional;
+
+    /** Named _group_ that the package is a member of. */
+    std::optional<std::string> packageGroup;
+
+    /** Force resolution is a given input or _flake reference_. */
+      std::optional<std::variant<std::string, nix::fetchers::Attrs>>
+    packageRepository;
+
+    /** Relative path to a `nix` expression file to be evaluated. */
+    std::optional<std::string> input;
+
+
+};  /* End struct `ManifestDescriptorRaw' */
+
+
+/* -------------------------------------------------------------------------- */
+
+/**
+ * @fn void from_json( const nlohmann::json        & jfrom
+ *                   ,       ManifestDescriptorRaw & desc
+ *                   )
+ * @brief Convert a JSON object to an @a flox::InputPreferences.
+ *
+ * @fn void to_json( nlohmann::json & jto, const ManifestDescriptorRaw & desc )
+ * @brief Convert an @a flox::resolver::ManifestDescriptorRaw to a JSON object.
+ */
+/* Generate to_json/from_json functions. */
+NLOHMANN_DEFINE_TYPE_NON_INTRUSIVE_WITH_DEFAULT( ManifestDescriptorRaw
+                                               , name
+                                               , version
+                                               , stability
+                                               , path
+                                               , absPath
+                                               , systems
+                                               , optional
+                                               , packageGroup
+                                               , packageRepository
+                                               , input
+                                               )
+
+
+/* -------------------------------------------------------------------------- */
+
+/**
+ * @brief Extend and remap fields from @a flox::resolver::PkgDescriptorRaw to
+ *        those found in a `flox` _manifest_.
+ */
+struct ManifestDescriptor {
```

Comment:
I'm not sure if this class is providing much value. `fillPkgQueryArgs` doesn't have much logic - it seems like it's mostly just copying values. Could we go straight from `ManifestDescriptorRaw` to `PkgQueryArgs`?

Reply:
This class holds the code which converts from one to the other.
Think of it as more of a closure/functor that transforms `ManifestDescriptorRaw -> PkgQueryArgs`.

It doesn't have to be a class all, it could just be a function `manifestDescriptorToQuery` or something with a bunch of helper functions; but for the purposes of testing, debugging, and organizing the docs a class felt ergonomic to me.